docs/proposals/device-crd.md

```diff
@@ -570,6 +570,24 @@ status:
 
 ### Validation
 [Open API v3 Schema based validation](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#validation) can be used to guard against bad requests.
+Invalid values for fields ( example string value for a boolean field etc) can be validated using this.
+In some cases , we also need custom validations (e.g create a device instance which refers to a non -existent device model ) .
+[Validation admission web hooks](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook) can be used to implement such custom validation rules.
+
+Here is a list of validations we need to support :
+
+#### Device Model Validations
+- Don't allow model creation if any `Required` fields are missing ( like property name , type, access mode etc.)
+- Property type can be only string or int64. Other types are not supported.
+- The Property access mode for a property can be one of `ReadWrite` or `ReadOnly`. Other access modes are not supported.
+- Property visitors are currently supported for bluetooth, modbus and opcua protocols only. Other protocol visitors are not supported.
+- The BluetoothArithmeticOperationType for Bluetooth read converter can have values one of `[Add, Subtract, Multiply, Divide]`
+- The ModbusRegisterType for Modbus visitor config can have values one of `[CoilRegister, DiscreteInputRegister, InputRegister, HoldingRegister]`. Other register types are not supported.
+
+#### Device Instance Validations
+- Don't allow device instance creation if any `Required` fields are missing ( like devicemodel ref , protocol config, twin property value, twin property name etc.)
+- Don't allow device instance creation if it refers to a device model reference which doesn't exist.
```

Comment:
Need to add a point that dont't allow device instance creation if any of the twin name is not present in devicemodel property list ?

Reply:
added.